components/fxa-client/README.md

```diff
@@ -0,0 +1,117 @@
+
+### Before using this component
+
+This component does not currently integrate with the [Glean SDK](https://mozilla.github.io/glean/book/index.html)
+and does not generate any telemetry, so you do not need to request a data-review before using this component.
+
+### Prerequisites
+
+To use this component, your application must be registered to [integrate with Firefox Accounts
+as an OAuth client](https://mozilla.github.io/ecosystem-platform/docs/process/integration-with-fxa)
+and have a unique OAuth `client_id`.
+
+You should also be familiar with how to integrate application-services components
+into an application on your target platform:
+* **Android**: integrate via the
+  [service-firefox-accounts](https://github.com/mozilla-mobile/android-components/tree/master/components/service/firefox-accounts/README.md)
+  component from android-components, which provides higher-level conveniences for state management, persistence,
+  and integration with other Android components.
+* **iOS**: start with the [guide to consuming rust components on
+  iOS](https://github.com/mozilla/application-services/blob/main/docs/howtos/consuming-rust-components-on-ios.md)
+  and take a look at the [higher-level Swift wrapper classes](./ios/FxAClient/).
+* **Other Platforms**: we don't know yet; please reach out on slack to discuss!
+
+### Core Concepts
+
+You should understand the core concepts of OAuth and the Firefox Accounts system
+before attempting to use this component. Please review the
+[Firefox Accounts Documentation](https://mozilla.github.io/ecosystem-platform/docs/features/firefox-accounts/fxa-overview)
+for more details.
+
+In particular, you should understand [the different types of auth token](
+https://github.com/mozilla/ecosystem-platform/pull/39)
+in the FxA ecosystem and how each is used, as well as how [OAuth scopes](
+https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/docs/oauth/scopes.md)
+work for accessing related services.
+
+### Component API
+
+For details on how to use this component, consult the [public interface definition](./src/fxa_client.udl) or view the generated Rust API documentation by running:
+
+```
+cargo doc --no-deps --open
+```
+
+## Working on the Firefox Accounts client
+
+### Prerequisites
+
+To effectively work on the FxA Client component, you will need to be familiar with:
+
+* Our general [guidelines for contributors](../../docs/contributing.md).
+* The [core concepts](#core-concepts) for users of the component, outlined above.
+* The way we use [uniffi-rs](https://github.com/mozilla/uniffi-rs) to generate API wrappers for multiple languages, such as Kotlin and Swift.
+
+### Overview
+
+This component uses [uniffi-rs](https://github.com/mozilla/uniffi-rs) to create its
+public API surface in Rust, and then generate bindings for Kotlin and Swift. The
+code is organized as follows:
+
+* The public API surface is implemented in [`./src/lib.rs`](./src/lib/rs), with matching
```

Comment:
Indeed. I think it would be good to pull it out into a common doc at some point, but IMHO that's premature until we get most of the components in a similar shape.